update-engine/src/buffer.rs

```diff
@@ -744,47 +784,83 @@ impl<'buf, S: StepSpec> EventBufferSteps<'buf, S> {
 pub struct EventBufferStepData<S: StepSpec> {
     step_info: StepInfo<NestedSpec>,
     sort_key: StepSortKey,
+    // XXX: nest_level and total_steps are common to each execution, but are
```

Comment:
I think ideally I would tie them together if they always exist together and get mutated together, but I wouldn't hold the PR for that.

Reply:
They always exist together but are immutable properties.